### DISCUSSION TITLE
How can i get the currently authenticated user?

### DISCUSSION THREAD

**Discussion by dotargz** [pocketbase/pocketbase#315] [Q&A] [answered]
I want to have users authenticate (using the JS SDK), and then be able to get data about them (probably via their ID) in future reloads of the page.
Is there a way to do this natively in pocketbase? Am I missing something? Should I just use local storage and store all the relevant data?

**Comment by ganigeorgiev** [comment, marked as answer]
You could use the `client.authStore` instance to extract the authenticated data:

```js
console.log(client.authStore.token); // output the token
console.log(client.authStore.model); // output the authenticated User or Admin model
```

More about the default `LocalAuthStore` you could read at https://github.com/pocketbase/js-sdk#authstore.

**Reply by andres-vizcaino** [reply]
If I want to have the user data from the server side (I am using Next JS), should I save it manually in the cookies?

**Reply by ganigeorgiev** [reply]
@andres-vizcaino For SSR there is no builtin solution yet, so it's up to the developer to create their own custom AuthStore and use cookies or any other storage mechanism. You could read more in issue https://github.com/pocketbase/js-sdk/issues/15.

_After v0.5.0 release, I'll try to research the available approaches more in depth and eventually will create an official SSRAuthStore._

**Reply by 5hanth** [reply]
any plans on SSRAuthStore yet?

**Reply by ganigeorgiev** [reply]
@5hanth You are commenting on a very old discussion and a lot of things has changed since then. Regarding the "SSRAuthStore" - the idea was discarded as it was difficult to generalize and will vary from one meta-framework to another and instead we added guides and simple cookie helpers to help with this scenarios. For more details please refer to https://github.com/pocketbase/js-sdk?tab=readme-ov-file#ssr-integration.

If you haven't already, please read also https://pocketbase.io/docs/how-to-use/.
